lib/compat/wordpress-6.1/class-wp-theme-json-6-1.php

```diff
@@ -650,37 +650,21 @@ public function get_stylesheet( $types = array( 'variables', 'styles', 'presets'
 
 		$stylesheet = '';
 
+		// Get variables.
 		if ( in_array( 'variables', $types, true ) ) {
 			$stylesheet .= $this->get_css_variables( $setting_nodes, $origins );
 		}
 
-		if ( in_array( 'styles', $types, true ) ) {
```

Comment:
Is the reason that this works because it means the styles now get output as part of the part of `presets` and `variables`? It appears that this results in the rules being output multiple times in the post editor:

<img width="436" alt="image" src="https://user-images.githubusercontent.com/14988353/190572907-8a674a07-cf66-4097-bfa7-58e7f7bf86f9.png">

I think this is occurring around here each time we call `gutenberg_get_global_stylesheet`: https://github.com/WordPress/gutenberg/blob/04d57b34b8574e1f25de0f0bc5862e3da045c6ac/lib/compat/wordpress-6.1/block-editor-settings.php#L63

Unfortunately this change seems to mean that the public API of `wp_get_global_stylesheet` when passed `array( 'variables' )` will unexpectedly also include styles. Instead of updating this check, should we update things one level up in `gutenberg_get_global_stylesheet`?

If it turns out we no longer need `base-layout-styles`, then we can still potentially remove that block below.

Reply:
I believe the fix might be that in `gutenberg_get_global_stylesheet` we'd remove the check on this line https://github.com/WordPress/gutenberg/blob/53d19c275c311d4912d136cb3638dafc511a0ef0/lib/compat/wordpress-6.1/get-global-styles-and-settings.php#L80 and instead that section would be (no `base-layout-styles`):

```php
	$tree                = WP_Theme_JSON_Resolver_Gutenberg::get_merged_data();
	$supports_theme_json = WP_Theme_JSON_Resolver_Gutenberg::theme_has_support();
	if ( empty( $types ) ) {
		$types = array( 'variables', 'styles', 'presets' );
	}
```

I haven't tried that out yet, but I believe basically we'd be removing any reference to `'base-layout-styles` in the repo.

Then, over in `block-editor-settings.php`, we'd need to ensure that we still request styles for classic themes, but that they would not be set to the `__unstableType` of `theme` (so that they're not considered part of the theme styles).

So, in this else statement, instead of using `base-layout-styles`, we might use something like the following here: https://github.com/WordPress/gutenberg/blob/04d57b34b8574e1f25de0f0bc5862e3da045c6ac/lib/compat/wordpress-6.1/block-editor-settings.php#L82

```
			$block_classes = array(
				'css'            => 'styles',
				'__unstableType' => 'base-styles', // setting this to something other than `theme` means that the editor will not consider these styles as part of "theme" styles.
				'isGlobalStyles' => true,
			);
```